activation/db.go

```
@@ -0,0 +1,129 @@
+package activation
+
+import (
+	"bytes"
+	"errors"
+	"github.com/davecgh/go-xdr/xdr2"
+	"github.com/spacemeshos/go-spacemesh/database"
+	"github.com/spacemeshos/go-spacemesh/mesh"
+)
+
+type ActivationDb struct {
+	//todo: think about whether we need one db or several
+	Atxs database.DB
+}
+
+func (db *ActivationDb) StoreAtx(atx mesh.ActivationTx) error {
+	if b, err := db.Atxs.Get(atx.Id().Bytes()); err == nil && len(b) > 0 {
+		// exists - how should we handle this?
```

Better return an error and don't swallow it, for now at least.

---

Still open.